### PR TITLE
Allow debug logs to be excluded from specified component

### PR DESCRIPTION
**Pull request by jnewbery**
setting `-debug` can lead to very noisy debug.logs with subcomponents filling up the log file. See for example https://travis-ci.org/bitcoin/bitcoin/jobs/216767286 where there are hundreds of libevent debug logs.

This commit adds an option to exclude certain components from debug logging. The usage is:

```
bitcoind -debug -debugexclude=<component1> -debugexclude=<component2> ...
```

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
utACK 3f1cbd38896cea8cea3095aad0a23254a39f651e

**Comment by JeremyRubin** [comment]
utACK 3f1cbd3.

Might be good to clean this up in another PR.

**Comment by jnewbery** [comment]
@JeremyRubin agree - this could be optimized so we don't do a `std::string.count()` every time `LogPrint()` is called. I think that's an improvement for a different PR. This doesn't make things particularly worse.

**Comment by jnewbery** [comment]
Closing until #9424 is merged.

**Comment by laanwj** [comment]
#9424 was merged

**Comment by jnewbery** [comment]
Reimplemented on top of the new logging categories infrastructure from #9424. As expected, it's about 10 lines of code.

Requesting review from @gmaxwell .

**Comment by jnewbery** [comment]
Sigh. Github won't let me force push once a PR has been closed. I'll open a new PR.

File: src/init.cpp [hunk 1]
```diff
@@ -914,13 +915,25 @@ bool AppInitParameterInteraction()
             for (const auto& cat : categories) {
                 uint32_t flag;
                 if (!GetLogCategory(&flag, &cat)) {
-                    InitWarning(strprintf(_("Unsupported logging category %s.\n"), cat));
```
Comment (laanwj) [inline src/init.cpp]: To not give the translators extra work, please try to use the same translation message as below. You could add a %s for the option.

**Comment by laanwj** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/10123/commits/39a70f26d9eb486762f553732e8ca5dbb754b72e apart from message nit.

File: src/init.cpp (hunk 1 above)
Comment (jnewbery) [inline src/init.cpp]: Good point. Changed.

**Comment by jnewbery** [comment]
@laanwj's nit fixed and squashed.

**Pull request merged**